tuxemon/mod_manager.py

```diff
@@ -0,0 +1,190 @@
+        if "dependencies" in meta:
+            dependencies = meta["dependencies"]
+
+            for pack in dependencies:
+                self.download_package(pack, release, repo, dont_extract=dont_extract, install_deps=False)
+
+                # Symlink deps
+                mainfolder = os.path.join(paths.BASEDIR, "mods", name)
+                depfolder = os.path.join(paths.BASEDIR, "mods", pack)
+                symlink_missing(mainfolder, depfolder)
+        else: pass
+
+    def extract(self, file, outfolder):
+        """Extracts the specified zip archive to the mods directory"""
+        with zipfile.ZipFile(file) as zip_:
+            zip_.extractall(path=os.path.join(paths.BASEDIR, "mods", outfolder))
+
+    def get_package_info(self, name, repo):
+        """Get specified package info. Always downloads the info from the server."""
+        with urllib.request.urlopen(repo + f"/api/packages/{name}") as packages:
+            return json.loads(packages.read().decode("UTF-8"))
+
+    def get_package_repo(self, name):
+        """Reads the origin of an package.
+        Returns None, if key 'mods' doesn't exist"""
+        for i in self.packages:
+            if i["name"] == name:
+                return i["repo"]
+                """
+                if "repo" in i: return i["repo"]
+                else: return None
+                """
+            else: continue
+
+    def write_package_to_list(self, path_to_folder, name):
+        """Writes specified package to the package list"""
+        # Write the absolute path to the list
+        with open(paths.USER_GAME_DATA_DIR + "/package.list", "w+") as file:
+            if not len(file.read()) == 0:
+                before = json.loads(file.read())
+            else:
+                before = {}
+
+            to_append = {name: path_to_folder}
+            after = {**before, **to_append}
+            file.write(
+                json.dumps(after, indent=4)
+            )
+
+    def read_package_from_list(self, name):
+        """Reads path of the specified mod"""
+        #try:
+        with open(paths.USER_GAME_DATA_DIR + "/package.list") as file:
+            data = file.read()
+            return json.loads(data)[name]
+
+    def remove_package_from_list(self, name):
+        """Removes specified package from the package list"""
+        # Write the absolute path to the list
+        with open(paths.USER_GAME_DATA_DIR + "/package.list", "r+") as file:
+            data = file.read()
+            if not len(data) == 0:
+                before = json.loads(data)
+            else:
+                raise ValueError("The package.list is empty.")
+
+            del before[name]
+            file.write(
+                json.dumps(before, indent=4)
+            )
+
+
+    def remove_package(self, name):
+        """Removes the local package"""
+        # Get the path
+        path = self.read_package_from_list(name)
+
+        shutil.rmtree(path, ignore_errors=True)
```

Comment:
this needs more checking so the name isn't something like `../../../../../home/`